backend/rules/graphprocessing/BuildGraph.go

```diff
@@ -0,0 +1,184 @@
+package graphprocessing
+
+import (
+	"fmt"
+	"github.com/Zeus-Labs/ZeusCloud/rules/types"
+)
+
+func PrintAdjList(dg *types.DisplayGraph) {
+	for nodeId, nodeList := range dg.AdjList {
+		fmt.Printf("Node Id: %+v \n", nodeId)
+		for _, node := range nodeList {
+			fmt.Printf("Node %+v \n", node)
+		}
+	}
+}
+
+// Returns boolean of if the ip range, security group, and compute
+// node.
+func extractIpRangeSecurityGroupCompute(dg *types.DisplayGraph,
```

Comment:
clearly state which pattern you are looking for - write it as ()-[]-()-() syntax as you would a match statement